learning-examples/mint_and_buy.py

```
@@ -0,0 +1,305 @@
+    print(f"  Bonding curve: {bonding_curve}")
+    print(f"  Associated bonding curve: {associated_bonding_curve}")
+    print(f"  Metadata: {metadata}")
+    print(f"  User ATA: {user_ata}")
+    print(f"  Creator vault: {creator_vault}")
+    
+    # Calculate buy parameters
+    # For pump.fun, we need to calculate expected tokens based on initial curve state
+    # Initial virtual reserves (from pump.fun constants)
+    initial_virtual_token_reserves = 1_073_000_000 * 10**TOKEN_DECIMALS
+    initial_virtual_sol_reserves = 30 * LAMPORTS_PER_SOL
+    initial_real_token_reserves = 793_100_000 * 10**TOKEN_DECIMALS
```

Ruff (0.11.9)

Local variable `initial_real_token_reserves` is assigned to but never used

Remove assignment to unused variable `initial_real_token_reserves`

(F841)